src/java/org/tensorics/core/tensor/AbstractTensorBuilder.java

```diff
@@ -100,18 +89,35 @@ private void checkIfContextValid(Position context2) {
 		}
 	}
 
-	
+	@Override
+	public void putAll(Tensor<E> tensor) {
+		this.putAll(Position.empty(), tensor);
+	}
 
 	@Override
 	public final void putAll(Position position, Tensor<E> tensor) {
 		checkNotNull(tensor, "The tensor must not be null!");
+		putAll(position, TensorInternals.mapFrom(tensor));
+	}
+
+	@Override
+	public void putAll(Map<Position, E> newEntries) {
+		putAll(Position.empty(), newEntries);
+	}
+
+	@Override

```

Comment:
Just a general comment ... do we need to expose this operation? It very much sound like what should rather by done by tensoric operations, where the first argument is a context-Position and the second is a Tensor.